articles/overview/overview.md

```diff
@@ -0,0 +1,81 @@
+
+#### Designer
+
+[Designers](xref:designers.about-designers) provide the interface that allow users to model what an Application should look and behave like while the designer itself also provides the structure for the model. The model can resemble a Tree-view or a Diagram of sorts. Each Designer exists to address a distinct aspect of the Application being developed. Examples are: modelling API services or what the underlying Database structure should look like, yet these Designers can also borrow information from one another so as to enrich the picture of what the Application looks like overall. Designers are themselves composed of Elements which are the building blocks for capturing user metadata.
+
+#### Designer Extension
+
+Existing Designers are be open for [extension](xref:designers.about-designer-extensions) which allows Module developers to add new Elements to a Designer from a different Module. This retains the idea of reuse and can provide users of Designers with a richer experience and with more ways to model what an Application. This should not be confused with Stereotypes.
+
+#### Stereotypes
+
+[Stereotypes](xref:stereotypes.about-stereotypes) offer additional ways to capture metadata from a user but not in the same way as Elements from Designers. They can only be applied on top of Elements (and/or their fields and associations) and offer a way to capture extra information about a given Designer Element.
+
+#### Designer API
+
+Not only do users of Intent Architect have an interface with which to model what the metadata should look like, Module developers will also get an easy to use generated API to query the modelled metadata for Template development. Example: [](xref:stereotypes.how-to-use-stereotypes#query-stereotypes-from-templates).
+
+### Code Generation
+
+Components associated with Code Generation will rely on the Designer API to supply it with metadata. This will then through some process transform the metadata into content that will be written to output files.
+
+#### Template
+
+Using a [Text Template](xref:templates.about-templates-csharp) containing control logic, metadata can be consistently transformed based on the format of the Text Template and the output will become the content of a text file. These file types can range from XML to programming languages source code as long as the output type is text based.
+
+#### Decorator
+
+Some Text Templates can be open for extension by exposing hook-in points for [Decorators](xref:templates.about-decorators) to supply additional content. Decorators offer a way to decouple certain logic from Text Templates. Decorators can also be installed through Modules that are separate from the Modules containing the Text Templates.
+
+### Advanced
+
+Components and infrastructural concerns additional to Metadata and Code Generation.
+
+#### Eventing Message
+
+Components can achieve a level of decoupling using the Publish / Subscriber pattern with Eventing Messages.
```

Comment:
Perhaps rather:

> A fully decoupled mechanism for modules to interact with each other through use of messages and the event dispatcher within the Software Factory, commonly used for fulfilling architectural infrastructural concerns.
>
> For example a template may need a registration to be performed with a system's dependency injection framework, without the template being aware of what dependency injection framework will ultimately be used. The template would simply publish a message saying what it needs, then multiple different modules (for example, AutoFac or Ninject modules) would be configured to listen to these messages and fulfil them making it so that regardless of which module is installed, everything still works.